man/ocitools-generate.1.md

```diff
@@ -84,6 +84,45 @@ inside of the container.
   The special *PATH* `host` removes any existing IPC namespace from the
   configuration.
 
+**--linux-cpu-shares**=CPUSHARES
+  Specifies a relative share of CPU time available to the tasks in a cgroup.
+
+**--linux-cpu-period**=CPUPERIOD
+  Specifies a period of time in microseconds for how regularly a cgroup's access to CPU resources should be reallocated (CFS scheduler only).
```

Comment:
On Fri, Aug 05, 2016 at 08:25:27PM -0700, hmeng-19 wrote:
> > +**--linux-cpu-period**=CPUPERIOD
> > +  Specifies a period of time in microseconds for how regularly a cgroup's access to CPU resources should be reallocated (CFS scheduler only).
> 
> @wking , @mrunalp , do correct me if I am wrong.
> I always thought the scheduling policy is a system-level config. It
> is decided when you built your kernel. Right?

That was my impression before poking around in the man pages today.
But see also [1], which has:

  Each process or thread is controlled by an associated scheduling
  policy and priority…

And [2] has:

  If at least one RT task is running, no other SCHED_OTHER task will
  be allowed to run in any CPU.

which makes it fairly clear that you can have threads/processes with
different schedulers running concurrently.  I'm not sure how that
works for the cgroup settings though.  Maybe you can set them all
(both real-time and CFS parameters) and the kernel will apply the
appropriate limits to each task depending on its policy?  In that
case, configs that specify settings for multiple schedulers are fair
game.

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sched.h.html
[2]: http://www.linuxjournal.com/article/3910